resolver-dns/src/test/java/io/netty/resolver/dns/DnsNameResolverTest.java

```diff
@@ -563,7 +510,156 @@ private static void queryMx(
+
+                        out.write(buf);
+                    }
+                };
+            }
+
+            @Override
+            public ProtocolDecoder getDecoder(IoSession session) throws Exception {
+                return new DnsUdpDecoder();
+            }
+
+            private static final class TestAAAARecordEncoder extends ResourceRecordEncoder {
+                private static final byte[] ADDRESS_BYTES = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1};
+                @Override
+                protected void putResourceRecordData(IoBuffer ioBuffer, ResourceRecord resourceRecord) {
+                    if (!"::1".equals(resourceRecord.get(DnsAttribute.IP_ADDRESS))) {
+                        throw new IllegalStateException("Only supposed to be used with IPV6 address of ::1");
+                    }
+                    // encode the ::1
+                    ioBuffer.put(ADDRESS_BYTES);
+                }
+            }
+        }
+
+        private static final class TestRecordStore implements RecordStore {
+            private static final int[] numbers = new int[254];
+            private static final char[] chars = new char[26];
+
+            static {
+                for (int i = 0; i < numbers.length; i++) {
+                    numbers[i] = i + 1;
+                }
+
+                for (int i = 0; i < chars.length; i++) {
+                    chars[i] =  (char) ('a' + i);
+                }
+            }
+
+            private static int index(int arrayLength) {
+                return Math.abs(ThreadLocalRandom.current().nextInt()) % arrayLength;
+            }
+
+            private static String nextDomain() {
+               return chars[index(chars.length)] + ".netty.io";
+            }
+
+            private static String nextIp() {
+                return ippart() + "." + ippart() + '.' + ippart() + '.' + ippart();
+            }
+
+            private static int ippart() {
+                return numbers[index(numbers.length)];
+            }
+
+            @Override
+            public Set<ResourceRecord> getRecords(QuestionRecord questionRecord) {
+                String name = questionRecord.getDomainName();
+                if (DOMAINS.contains(name)) {
+                    ResourceRecordModifier rm = new ResourceRecordModifier();
+                    rm.setDnsClass(RecordClass.IN);
+                    rm.setDnsName(name);
+                    rm.setDnsTtl(100);
+                    rm.setDnsType(questionRecord.getRecordType());
+
+                    switch (questionRecord.getRecordType()) {
+                    case A:
+                        do {
+                            String ip = nextIp();
+                            rm.put(DnsAttribute.IP_ADDRESS, ip);
+                        } while (ThreadLocalRandom.current().nextBoolean());
+                        break;
+                    case AAAA:
+                        do {
+                            rm.put(DnsAttribute.IP_ADDRESS, "::1");
```

Comment:
Is this a PITA to add diversity for bcz the dns server doesn't really support AAAA records?